ParkingLot/SmartParkingBoy.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ParkingLot
+{
+    public class SmartParkingBoy : ParkingBoy
+    {
+        public SmartParkingBoy(List<ParkingLot> parkingLotList) : base(parkingLotList)
+        {
+        }
+
+        public override ParkingTicket ParkCar(ICar car, out string errorMessage)
+        {
+            if (!IsInputCarValid(car))
+            {
+                errorMessage = null;
+                return null;
+            }
+
+            if (this.ParkingLotList.Where(parkingLot => parkingLot.HasPosition()).ToList().Count == 0)
+            {
+                errorMessage = "Not enough position.";
+                return null;
+            }
+
```

Comment:
if every boy need to check position once, should this logic be placed inparkinglot?
anyway the design now also work